chunks/modelevaluation.ipynb

```diff
@@ -0,0 +1,935 @@
+    "\n",
+    "* We expect $\\chi^2_{\\rm min}$ to be distributed as $\\chi^2$ for $N_{\\rm dof} = (N_{\\rm data}-2)$ degrees of freedom. \n",
+    "\n",
+    "* We can ask for the probability of getting a dataset that gives our value of $\\chi^2_{\\rm min}$ or worse (greater), given that our dataset and all the hypothetical ones were generated from our model (the \"null hypothesis\"). This one-sided integral probability is the \"p-value\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Chi-squared hypothesis testing\n",
+    "\n",
+    "* The \"p-value\" gives the probability of getting a value of $\\chi^2_{\\rm min}$, or higher, by chance, given that the model is true. \n",
+    "\n",
+    "* If the p-value is less than some significance level of our choosing (e.g. 0.05), then the null hypothesis (our straight line model) would be rejected at this level.\n",
+    "\n",
+    "* We can compute the p-value assuming a chi-squared distribution using `scipy.stats`:\n",
+    "```python\n",
+    "import scipy.stats\n",
+    "chisq = scipy.stats.chi2(Ndof)\n",
+    "pvalue = chisq.sf(chisq_min)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Chi-squared hypothesis testing\n",
+    "\n",
+    "* In our case, we would \"reject the null hypothesis at the $10^{-10}$ significance level\"\n",
+    "```\n",
+    "chisq_min =  104.2 p-value =  1.03e-10\n",
+    "```\n",
+    "\n",
+    "* The \"reduced chi-squared\", $\\hat{\\chi}^2 = \\chi^2_{\\rm min} / N_{\\rm dof}$, is often used by astronomers to quantify goodness of fit - but note that you need to know the number of degrees of freedom separately to be able to interpret it.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Fisher's Approximation to the Chi-squared Distribution\n",
+    "\n",
+    "* A useful, quick way to make sense of  $\\chi^2_{\\rm min}$ and $N_{\\rm dof}$ values is to use Fisher's Gaussian approximation to the chi-squared distribution: \n",
+    "\n",
+    "$\\;\\;\\;\\;\\;\\sqrt{2\\chi^2_{\\rm min}} \\sim \\mathcal{N}\\left( \\sqrt{2 N_{\\rm dof}-1}, 1 \\right)$\n",
+    "\n",
+    "* The difference between $\\sqrt{2\\chi^2_{\\rm min}}$ and $\\sqrt{2 N_{\\rm dof}-1}$ is the \"number of sigma\" ($n_{\\sigma}$) we are away from a good fit.\n",
+    "\n",
+    "* In our case, the MLE model is about 7-sigma away from being a good fit."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Exercise: $\\chi^2$ intuition\n",
+    "\n",
+    "Write a short function that takes in values of `chisq` and `Ndof` and use it to populate this table:\n",
```

Comment:
OK fair enough: I'll give them the code and they can fiddle around with the numbers a little